crates/api_common/src/request.rs

```diff
@@ -63,47 +64,45 @@ pub async fn fetch_link_metadata(url: &Url, context: &LemmyContext) -> LemmyResu
     .await?
     .error_for_status()?;
 
-  // In some cases servers send a wrong mime type for images, which prevents thumbnail
-  // generation. To avoid this we also try to guess the mime type from file extension.
-  let content_type = mime_guess::from_path(url.path())
-    .first()
-    // If you can guess that its an image type, then return that first.
-    .filter(|guess| guess.type_() == mime::IMAGE)
-    // Otherwise, get the content type from the headers
-    .or(
-      response
-        .headers()
-        .get(CONTENT_TYPE)
-        .and_then(|h| h.to_str().ok())
-        .and_then(|h| h.parse().ok()),
-    );
-
-  let opengraph_data = {
-    // if the content type is not text/html, we don't need to parse it
-    let is_html = content_type
-      .as_ref()
-      .map(|c| {
-        (c.type_() == mime::TEXT && c.subtype() == mime::HTML)
-      ||
-      // application/xhtml+xml is a subset of HTML
-      (c.type_() == mime::APPLICATION && c.subtype() == "xhtml")
-      })
-      .unwrap_or(false);
-    if !is_html {
-      Default::default()
-    } else {
+  let mut content_type: Option<Mime> = response
```

Comment:
There is no reason to make content_type mut here. Add an `.or(...` to L72

Reply:
I moved the `mime_guess` to there, but on L99 I rebind `content_type` and I don't see how I'd be able to move that up.

Reply:
I couldn't remove the mut either, because of no async in closures, and I couldn't move up the octet check because of the response cloning issues 🤷 . But I was able to clean up some things:

https://github.com/flamingo-cant-draw/lemmy/pull/1